### ISSUE TITLE
should implement `readbytes`

### ISSUE THREAD

**Issue by JeffBezanson** [JuliaLang/GZip.jl#28]
I hereby apologize for the `readall` function. I think GZip should implement `readbytes` instead, and inherit `readall(::IO)` that just wraps the result of `readbytes` in a string.

**Comment by quinnj** [comment]
Yeah, it should really implement `readbytes!` or even better would be `readavailable`; as-is, the only standard IO function implemented is `read(::GZipStream, ::Type{UInt8})`, which means the majority of cases where a `GZipStream` is used generically falls back to byte-by-byte IO, which is probably terrible for performance.
